.github/workflows/contract.yml

```diff
@@ -108,3 +108,64 @@ jobs:
           name: test-results.xml
           path: test-results.xml
           if-no-files-found: error
+
+  leap-integration:
+    runs-on: ubuntu-20.04
+    name: EOS EVM Leap Integration Tests
+    env:
+      CC: gcc-10
+      CXX: g++-10
+      DCMAKE_BUILD_TYPE: 'Release'
+
+    steps:
+      - name: Authenticate
+        id: auth
+        uses: AntelopeIO/github-app-token-action@v1
+        with:
+          app_id: ${{ secrets.TRUSTEVM_CI_APP_ID }}
+          private_key: ${{ secrets.TRUSTEVM_CI_APP_KEY }}
+
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: 'recursive'
+          token: ${{ steps.auth.outputs.token }}
+
+      - name: Download CDT
+        uses: AntelopeIO/asset-artifact-download-action@v2
+        with:
+          owner: AntelopeIO
+          repo: cdt
+          target: 'v3.1.0'
+          prereleases: false
+          file: 'cdt_.*amd64.deb'
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install CDT
+        run: sudo apt-get install -y ./cdt*.deb
+
+      - name: Build EOS EVM Contract
+        run: .github/workflows/build-contract.sh
+        env:
+          DWITH_TEST_ACTIONS: ${{ matrix.DWITH_TEST_ACTIONS }}
+
+      - name: Download Leap - dev binary
+        uses: AntelopeIO/asset-artifact-download-action@v2
+        with:
+          owner: AntelopeIO
+          repo: leap
+          target: 'v4'
+          prereleases: false
+          file: 'leap-dev.*(x86_64|amd64).deb'
+          container-package: experimental-binaries
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Leap
+        run: sudo apt-get install -y ./leap*.deb
+
+      - name: Test Leap Integration
+        run: |
+          pip install web3
+          export PYTHONPATH=/usr/local/lib/python3/dist-packages
```

Comment:
Did something change? afaik all this should be installed in `/usr` when built in CI
https://github.com/AntelopeIO/leap/blob/main/.github/workflows/build_base.yaml#L80

Reply:
That's what I'm endeavoring to figure out.  I just did build using the `pinned_build.sh` script and then installed the `lead-dev` pkg it created. Using `dpkg -L leap-dev` it appeared that `TestHarness` was installed in `/usr/local/lib/python3/dist-packages/TestHarness`.  But it still isn't finding the module there either.

Reply:
Yeah, `pinned_build.sh` does `/usr/local` by default (but not in CI where it's also `/usr`)

Reply:
Noted. Updated the location where it installed.

Reply:
Why do we need to `export PYTHONPATH` in this case, where the test in leap doesn't need to?
https://github.com/AntelopeIO/leap/blob/main/.github/workflows/build.yaml#L98

Reply:
I'm trying to chase that down.  My current theory is that the symlink isn't being created to put the `TestHarness` into the `python3/dist-packages`.  This build is running `cmake version 3.26.4`. Not sure which version is being used over in the `leap` workflow since there is a switch in https://github.com/AntelopeIO/leap/blob/78c1b62a76b37ac3531e526b8434c1ce54a451c2/CMakeLists.txt#L252

Reply:
`dpkg -L leap-dev | grep TestHarness` in this workflow is not showing the symlink:

```
/usr/share/leap_testing/tests/TestHarness
/usr/share/leap_testing/tests/TestHarness/Cluster.py
/usr/share/leap_testing/tests/TestHarness/Node.py
/usr/share/leap_testing/tests/TestHarness/TestHelper.py
/usr/share/leap_testing/tests/TestHarness/WalletMgr.py
/usr/share/leap_testing/tests/TestHarness/__init__.py
/usr/share/leap_testing/tests/TestHarness/core_symbol.py
/usr/share/leap_testing/tests/TestHarness/depresolver.py
/usr/share/leap_testing/tests/TestHarness/interfaces.py
/usr/share/leap_testing/tests/TestHarness/launch_transaction_generators.py
/usr/share/leap_testing/tests/TestHarness/libc.py
/usr/share/leap_testing/tests/TestHarness/logging-template.json
/usr/share/leap_testing/tests/TestHarness/logging.py
/usr/share/leap_testing/tests/TestHarness/queries.py
/usr/share/leap_testing/tests/TestHarness/testUtils.py
/usr/share/leap_testing/tests/TestHarness/transactions.py
```

https://github.com/eosnetworkfoundation/eos-evm/actions/runs/5613879938/job/15210866007

Reply:
Although local build (note: using `pinned_build.sh` to create `leap-dev`) with `cmake version 3.22.1` does:

```
/usr/local/share/leap_testing/tests/TestHarness
/usr/local/share/leap_testing/tests/TestHarness/Cluster.py
/usr/local/share/leap_testing/tests/TestHarness/Node.py
/usr/local/share/leap_testing/tests/TestHarness/TestHelper.py
/usr/local/share/leap_testing/tests/TestHarness/WalletMgr.py
/usr/local/share/leap_testing/tests/TestHarness/__init__.py
/usr/local/share/leap_testing/tests/TestHarness/core_symbol.py
/usr/local/share/leap_testing/tests/TestHarness/depresolver.py
/usr/local/share/leap_testing/tests/TestHarness/interfaces.py
/usr/local/share/leap_testing/tests/TestHarness/launch_transaction_generators.py
/usr/local/share/leap_testing/tests/TestHarness/libc.py
/usr/local/share/leap_testing/tests/TestHarness/logging-template.json
/usr/local/share/leap_testing/tests/TestHarness/logging.py
/usr/local/share/leap_testing/tests/TestHarness/queries.py
/usr/local/share/leap_testing/tests/TestHarness/testUtils.py
/usr/local/share/leap_testing/tests/TestHarness/transactions.py
/usr/local/lib/python3/dist-packages/TestHarness
```

So from that perspective it doesn't look like a cmake version problem.

Reply:
It appears that this is due to the `postinst` script relying on `CMAKE_INSTALL_FULL_LIBDIR` in `v4.0.4` which sets the path for the symlink incorrectly to: `ln -s ../../../share/leap_testing/tests/TestHarness /usr/lib/x86_64-linux-gnu/python3/dist-packages/TestHarness`.

This has been fixed in `main`: https://github.com/AntelopeIO/leap/blob/main/scripts/postinst